javascript/Spread_operator.md

```diff
@@ -0,0 +1,198 @@
+
+```javaScript
+const obj = {
+    name: 'AppDividend',
+    author: 'Krunal Lathiya'
+};
+const combined = {
+    ...obj,
+    age: 25
+};
+console.log(combined);
+```
+
+
+### Arrays
+
+
+#### normal array concat() method 
+
+```javaScript
+let arr = [1,2,3]; 
+let arr2 = [4,5]; 
+  
+arr = arr.concat(arr2); 
+  
+console.log(arr); // [ 1, 2, 3, 4, 5 ] 
+```
+
+#### spread operator doing the concat job 
+
+```javaScript
+let arr = [1,2,3]; 
+let arr2 = [4,5]; 
+  
+arr = [...arr,...arr2]; 
+console.log(arr); // [ 1, 2, 3, 4, 5 ] 
+```
+
+#### spread operator for copying 
+
+```javaScript 
+let arr = ['a','b','c']; 
+let arr2 = [...arr]; 
+  
+console.log(arr); // [ 'a', 'b', 'c' ] 
+  
+arr2.push('d'); //inserting an element at the end of arr2 
+  
+console.log(arr2); // [ 'a', 'b', 'c', 'd' ] 
+console.log(arr); // [ 'a', 'b', 'c' ] 
+```
+
+#### copying without the spread operator 
+
+```javaScript
+let arr = ['a','b','c']; 
+let arr2 = arr; 
+  
+console.log(arr2); // [ 'a', 'b', 'c' ] 
+```
+[From geeksforgeeks](https://www.geeksforgeeks.org/javascript-spread-operator/)
+
+### Spread in object literals
+
+It copies own enumerable properties from a provided object onto a new object.In object literals merging of objects is now possible using a shorter syntax than Object.assign().
+
+```javaScript
+const obj1 = { foo: 'bar', x: 42 };
+const obj2 = { foo: 'baz', y: 13 };
+
+const clonedObj = { ...obj1 };
+// Object { foo: "bar", x: 42 }
+
+const mergedObj = { ...obj1, ...obj2 };
+// Object { foo: "baz", x: 42, y: 13 }
+```
+
+## Array slice method
+
+The slice() method returns the selected elements in an array, as a new array object.The slice() method selects the elements starting at the given start argument, and ends at, but does not include, the given end argument.
+The **syntax** of slice method is `array.slice(start, end)`.
+
+### The Slice method takes 2 arguments.
+
+**1st Argument:** Specifies from where the selection should be started.
+
+For Example:
+
+```
+var arr1 = [1,5,8,9];
+arr1.slice(1); // [5,8,9]
+From the first index (5) it will return the elements.
+```
+
+**2nd Argument:** Specifies at which level the endpoint should be. If you didn’t put this in the parenthesis while calling the slice method, it will return the elements from the starting index to the end of the array.
+
+```
+var arr1 = [1,5,8,9];
+console.log(arr1.slice(1,3));
+//[ 5, 8 ]
+```
+
+If you put a negative number while calling, the selection will be selected from the end of the array.
+
+```
+var arr1 = [1,5,8,9];
+console.log(arr1.slice(-2));
+//[ 8, 9 ]
+```
+
+Note: Slice always returns the selected elements from the array.
+
+Slice won’t change the array. The array remains intact. See the below Example:
+
+```
+var arr1 = [1,5,8,9];
+arr1.slice(2);
+console.log(arr1);
+// [ 1, 5, 8, 9 ]
+```
+
+Even if you made some changes to the array it won’t affect it. It will return the original array as it is at the start.
```

Comment:
This is excellent 🌟
Can you also mention with an example that for this reason, you need to assign the sliced array to a variable if you want to keep it?

Reply:
Also, please put in a word about this AC: "how this relates to following an immutable coding style to always create new objects and never modify arrays in place"

Reply:
You reiterated that it will not change, but please address: "how this relates to following an immutable coding style" and why you should " always create new objects and never modify arrays in place".  Not just that it is being done, but why.

Reply:
@urbashi123 Please address this.